### PR TITLE
very very minor tweaks while looking through the code base

### PR DESCRIPTION
**Pull request by olliethomas**
There's nothing in this PR that I'm passionate about if you're not happy with any of it. The only change that I think might be controversial is using augmented assignment as I know that has caused problems before - I'll revert this if you'd prefer.

### CODE REVIEW

File: whaledo/transforms.py [hunk 1]
```diff
@@ -253,7 +251,7 @@ def __init__(
         *,
         global_transform_1: ImageTform,
         global_transform_2: Optional[ImageTform] = None,
-        local_transform: LT = None,
+        local_transform: Optional[ImageTform] = None,
```
Comment (MylesBartlett) [inline whaledo/transforms.py]: Using a generic makes the transform displayable in the type, which I like

File: whaledo/transforms.py [hunk 2]
```diff
@@ -231,11 +230,10 @@ def __len__(self) -> int:
     def __add__(self, other: Self) -> Self:
         copy = gcopy(self, deep=False)
         copy.global_views = copy.global_views + other.global_views
-        if copy.local_views is None:
```
Comment (MylesBartlett) [inline whaledo/transforms.py:234]: other is required to be of the same type (Self) so whether copy or other is used for the checks is irrelevant here

File: whaledo/transforms.py [hunk 3]
```diff
@@ -175,7 +174,7 @@ def num_sources(self) -> int:
 @dataclass
 class MultiCropOutput(InputContainer[MultiViewPair]):
     global_views: MultiViewPair
-    local_views: Tensor
+    local_views: Optional[Tensor]
```
Comment (MylesBartlett) [inline whaledo/transforms.py:177]: Why Optional?

Comment (MylesBartlett) [reply whaledo/transforms.py:177]: Okay because of the __add__ method which is outdated now -- MultiViewPair is the form taken when local_views is None

File: whaledo/models/base.py [hunk 4]
```diff
@@ -74,10 +71,10 @@ def predict(
             db_size -= 1
 
         k = min(k, db_size)
-        scores, topk_inds = probs.topk(dim=1, k=k, sorted=sorted)
+        scores, topk_inds = probs.topk(dim=1, k=k, sorted=sorted_)
         mask = self.threshold_scores(scores=scores)
         n_retrieved_per_query = mask.count_nonzero(dim=1)
-        mask_inds = mask.nonzero(as_tuple=True)
+        mask_inds = mask.nonzero(as_tuple=True)  # I get unexpected argument `as_tuple` here.
```
Comment (MylesBartlett) [inline whaledo/models/base.py]: It's very much a valid argument -- I use it extensively

File: whaledo/models/artifact.py [hunk 5]
```diff
@@ -73,7 +72,7 @@ def download_artifact(root: Union[str, Path], *, run: Union[Run, RunDisabled], n
     artifact = run.use_artifact(full_name)
     root = Path(root)
     LOGGER.info(f"Downloading artifact from '{full_name}' to {root.resolve()}")
-    artifact.download(root=root)
+    artifact.download(root=str(root))
```
Comment (MylesBartlett) [inline whaledo/models/artifact.py]: Seems to work fine with a Path object

File: whaledo/data/datamodule.py [hunk 6]
```diff
@@ -55,7 +55,7 @@ def train_dataloader(
         self, *, shuffle: bool = False, drop_last: bool = False, batch_size: Optional[int] = None
     ) -> CdtDataLoader[SampleType]:
         batch_size = self.train_batch_size if batch_size is None else batch_size
-        batch_sampler = None
+        batch_sampler: Optional[Union[QueryKeySampler, SequentialBatchSampler]] = None
```
Comment (MylesBartlett) [inline whaledo/data/datamodule.py]:
```suggestion
        batch_sampler: Optional[BatchSamplerBase] = None
```

File: whaledo/algorithms/loss.py [hunk 7]
```diff
@@ -171,7 +177,7 @@ def supcon_loss(
     )
     logits = anchors[selected_rows] @ candidates_t.T
     # Apply temperature-scaling to the logits.
-    logits = logits / temperature
+    logits /= temperature
```
Comment (MylesBartlett) [inline whaledo/algorithms/loss.py]: in-place operations corrupt the computational graph

File: whaledo/relay.py [hunk 8]
```diff
@@ -116,7 +116,6 @@ def run(self, raw_config: Dict[str, Any]) -> None:
         if self.save_model and (not logger.experiment.offline):
             if self.save_best and cast(str, best_model_path := checkpointer.best_model_path):
                 self.log(f"Loading best model from checkpoint '{best_model_path}'.")
-                alg.load_from_checkpoint
```
Comment (tmke8) [inline whaledo/relay.py:119]: Maybe this was a bug

Comment (MylesBartlett) [reply whaledo/relay.py:119]: Not a bug, just the vestiges of debugging

File: whaledo/models/artifact.py (hunk 5 above)
Comment (olliethomas) [reply whaledo/models/artifact.py]: yeah, I suspect it's just mistyped. I don't see a reason why Path would be invalid.

File: whaledo/relay.py (hunk 8 above)
Comment (MylesBartlett) [reply whaledo/relay.py:119]: It's dead code, nothing that would screw anything up by addition or omission

File: whaledo/models/base.py (hunk 4 above)
Comment (olliethomas) [reply whaledo/models/base.py]: do you reckon it's a version issue? or just bad typing?

Comment (olliethomas) [reply whaledo/models/base.py]: `def nonzero(self): # real signature unknown; restored from __doc__`
ok - so it must be a generated type stub
